Release 0.4.1 (bellpepper only)

huitseeker opened the pull request:
Ports #63 to main and releases bellpepper at 0.4.1.
```
huitseeker@lightning.localdomain➜bellpepper/crates/bellpepper(release-0.4.1)» cargo semver-checks                                                                                                                                    [12:45:58]
     Parsing bellpepper v0.4.1 (current)
     Parsing bellpepper v0.4.0 (baseline, cached)
    Checking bellpepper v0.4.0 -> v0.4.1 (minor change)
   Completed [   0.011s] 45 checks; 45 passed, 6 unnecessary
```
Differences with released 0.4.0 include #63 #60 #58 #55.

Note that #58 removes code introduced in #55 and so the conjunction of the two has no difference with the last released version w.r.t negative numbers.

porcuquine reviewed (approved):
I am approving, but even with the noted cancellation of changes, it's not clear to me that everything in the description is yet represented in the change set here.

huitseeker commented:
@porcuquine Does this allay your concerns?
```
huitseeker@lightning.localdomain➜~/tmp/bellpepper(release-0.4.1)» git log release/all/0.4.0..HEAD                                                                                                                                    [14:21:04]
commit 0057b8ff8df59d7356d9a59266e0b7dae6b7eaf3 (HEAD -> release-0.4.1, origin/release-0.4.1)
Author: François Garillot <francois@garillot.net>
Date:   Tue Dec 5 12:44:18 2023 -0500

    chore: Update bellpepper crate to version 0.4.1

    - Updated `bellpepper` crate version to `0.4.1`.

commit 269f88f51c79186f5e7391b7dc3051406bd7b056
Author: Hanting Zhang <hantingz@usc.edu>
Date:   Wed Nov 15 12:07:40 2023 -0800

    pre-populate one (#63)

commit cdff4705cd14c580ddc305a9bfe5821468b4decf (main)
Author: Gabriel Barreto <gabriel.aquino.barreto@gmail.com>
Date:   Tue Nov 14 15:32:16 2023 -0300

    some WitnessCS utility functions (#60)

commit 88d8af2a1a8c73247c96f9a1f213ca6e26001f64 (gabriel-barrett/main)
Author: François Garillot <4142+huitseeker@users.noreply.github.com>
Date:   Sun Nov 12 11:00:16 2023 -0500

    refactor: Remove negative number definition from bellpepper crate (#58)

    - Eliminated the `negative_num` feature from the `bellpepper` crate.
    - Deleted the `negative_num` module from the gadget subset in the `bellpepper` crate.
    - definition was unreleased and unused.

commit 10e5b56f387d7a918353c1258f64f021915a942d
Author: François Garillot <4142+huitseeker@users.noreply.github.com>
Date:   Fri Nov 10 21:32:26 2023 -0500

    feat: Enhance boolean operations in bellpepper crate (#55)

    * feat: Enhance boolean operations in bellpepper crate

    - Introduce a new module for variadic boolean utility functions,
        - Implement new variadic versions of boolean OR and AND operations, i.e., `or_v`, `or_v_unchecked_for_optimization`, `and_v`.
        - Integrate boolean `or!` and `and!` macros and add allocation function to verify zero values & arity.
    - Conduct tests for the added and/or macros and various utility functions.

    * feat: add negative num definition

    - Implemented a new `negative_num` module in the `bellpepper` crate's `gadgets` directory, which includes the function `allocate_is_negative` for determining number negativity.
    - Overhauled the module declarations order in `gadgets/mod.rs` for improved structuring.

    * fix: make alloc_num_is_zero private

commit 46a3cc4ee72410bd98e67a5f65f96c80f8947942
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Tue Nov 7 08:30:34 2023 -0500

    chore: Release (#51)

    Co-authored-by: github-actions <github-actions@github.com>
huitseeker@lightning.localdomain➜~/tmp/bellpepper(release-0.4.1)» git lg origin/main..HEAD                                                                                                                                           [14:21:23]
* 0057b8f - (HEAD -> release-0.4.1, origin/release-0.4.1) chore: Update bellpepper crate to version 0.4.1 (2 hours ago)<François Garillot>
* 269f88f - pre-populate one (#63) (2 hours ago)<Hanting Zhang>
```

porcuquine commented:
My confusion was caused by this PR making some small changes and also releasing — so the description describes changes (to be released) that are not in the PR itself, though the PR does make some changes. I now understand the state.
